File: unix/tooltip.c
#include "uipriv_unix.h"

void uiControlSetTooltip(uiControl *c, const char *tooltip) {
	if (tooltip == NULL) {
		gtk_widget_set_has_tooltip(GTK_WIDGET(uiControlHandle(c)), FALSE);
	} else {
		gtk_widget_set_tooltip_text(GTK_WIDGET(uiControlHandle(c)), tooltip);
	}
}

File: darwin/tooltip.m
#include "uipriv_darwin.h"

// https://developer.apple.com/documentation/appkit/nsview/1483541-tooltip?language=objc

void uiControlSetTooltip(uiControl *c, const char *tooltip) {
	NSView *view = (NSView *)uiControlHandle(c);
	if (tooltip == NULL) {
		view.toolTip = nil;
	} else {
		view.toolTip = uiprivToNSString(tooltip);
	}
}
